src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIGoHome.java

```diff
@@ -58,7 +58,16 @@ public EntityAIGoHome(final EntityCitizen citizen)
     @Override
     public boolean shouldExecute()
     {
-        return citizen.getDesiredActivity() == DesiredActivity.SLEEP && !citizen.getCitizenColonyHandler().isAtHome();
+        boolean areBarbariansInColony;
+        if(citizen.getCitizenColonyHandler().getColony().getBarbManager().getHorde().size() > 0)
+        {
+            areBarbariansInColony = true;
+        }
+        else
+        {
+            areBarbariansInColony = false;
+        }
+        return citizen.getDesiredActivity() == DesiredActivity.SLEEP && !citizen.getCitizenColonyHandler().isAtHome() || areBarbariansInColony && !citizen.getCitizenColonyHandler().isAtHome();
```

Comment:
I think this can be:

+        return !citizen.getCitizenColonyHandler().isAtHome()  && (citizen.getDesiredActivity() == DesiredActivity.SLEEP  || areBarbariansInColony)